DSCResources/MSFT_xDnsARecord/MSFT_xDnsARecord.psm1

```diff
@@ -0,0 +1,73 @@
+﻿function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Zone,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $Target
+    )
+
+    Write-Verbose "Looking up DNS record for $Name in $Zone"
+    return (Get-DnsServerResourceRecord -ZoneName $Zone -Name $Name)
```

Comment:
You aren't returning a hashtable.

Example code:

    $record = Get-DnsServerResourceRecord -ZoneName $Zone -Name $Name
    return @{
       Name = $Name
       Zone = $Zone
      Target =  $record.HostName
    }